### PR TITLE
chore(bluejay-parser): Disallow reserved keyword as enum value

### LINKED ISSUE #31: Parse error for enum value definitions `true`, `false`, and `null`
Per the [spec](https://spec.graphql.org/draft/#sel-HAFdTDCAEAAAAEBCAE8oE) enum values must be:
> _Name_ but not `true` or `false` or `null`

We should return a parse error when parsing enum value definitions where the name is invalid

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Resolves #31 

This disallows reserved keywords as an enum value definition.

### CODE REVIEW

**Review by adampetro — COMMENTED** [review]
Can you add an error test case [here](https://github.com/Shopify/bluejay/tree/536a324711e11949e7da67109f97f0b6bf7f9bd6/bluejay-parser/tests/test_data/schema_definition/error)?

File: bluejay-parser/src/ast/parse_error.rs [hunk 1]
```diff
@@ -3,6 +3,10 @@ use crate::Span;
 
 #[derive(Debug)]
 pub enum ParseError {
+    UnexpectedEnumValue {
```
Comment (swalkinshaw) [inline bluejay-parser/src/ast/parse_error.rs:6]:
```suggestion
    InvalidEnumValue {
```

File: bluejay-parser/src/ast/parse_error.rs [hunk 2]
```diff
@@ -26,6 +30,14 @@ pub enum ParseError {
 impl From<ParseError> for Error {
     fn from(val: ParseError) -> Self {
         match val {
+            ParseError::UnexpectedEnumValue { span, value } => Self::new(
+                "Parse error",
+                Some(Annotation::new(
+                    format!("{value} is not an allowed enum-value"),
```
Comment (swalkinshaw) [inline bluejay-parser/src/ast/parse_error.rs:36]:
```suggestion
                    format!("{value} is not an allowed enum value"),
```

**Review by swalkinshaw — APPROVED** [review]
Two suggestions for naming

**Pull request merged**